# Metadata

```yaml
Course:  DS 5100
Module:  09 Python Packages
Topic:   HW Package Booklover
Author:  R.C. Alvarado (adapted)
Date:    7 July 2023 (revised)
```

# Student Info

* Name: Michael Amadi
* Net UD: mxg9xv
* URL of this file in GitHub: https://github.com/Datascifer/hwm09p/blob/main/M09-HW.ipynb

# Instructions

In your **private course repo on Rivanna**, use this Jupyter notebook and the data file described to write code that performs the tasks below. 

Save your notebook in the `M09` directory.

Remember to add and commit these files to your repo. 

Then push your commits to your repo on GitHib.

Be sure to fill out the **Student Info** block above.

To submit your homework, save your results as a PDF and upload it to GradeScope. More information about how to create the PDF for this assignment are included at the end of this document.

**TOTAL POINTS: 8**

# Overview

Follow the following recipe we used in class to package the code you wrote for `HW08` -- `booklover.py` and `booklover_test.py`.

* Create a new git repo for your package.
* Create and edit the required files and directories for your package and move the booklover modules there.
* Stage, commit, and push all the files you've created.
* Install your package with pip.
* Outside of your package dir, write a script to test your method.

Put this notebook in your repo. This will allow you to execute bash commands and capture the outpunt directly in the notebook.

**TOTAL: 8 POINTS**

# Tasks

## Task 1

(5 points)

Show the directory structure of your repo by running this command from the root of your repo:

In [1]:
!ls -lR

.:
total 36
drwx--S--- 4 mxg9xv users 1024 Apr  1 05:32 build
drwx--S--- 3 mxg9xv users 2560 Apr  1 06:13 demo
drwx--S--- 2 mxg9xv users 2048 Apr  1 05:32 demo.egg-info
-rw------- 1 mxg9xv users 1067 Apr  1 04:41 LICENSE
-rw------- 1 mxg9xv users 7073 Apr  1 06:14 M09-HW.ipynb
-rw------- 1 mxg9xv users    8 Apr  1 04:41 README.md
-rw------- 1 mxg9xv users  337 Apr  1 04:50 setup.py

./build:
total 8
drwx--S--- 2 mxg9xv users   0 Apr  1 05:32 bdist.linux-x86_64
drwx--S--- 3 mxg9xv users 512 Apr  1 05:32 lib

./build/bdist.linux-x86_64:
total 0

./build/lib:
total 4
drwx--S--- 2 mxg9xv users 1536 Apr  1 05:32 demo

./build/lib/demo:
total 16
-rw------- 1 mxg9xv users 1198 Apr  1 04:47 booklover.py
-rw------- 1 mxg9xv users 1782 Apr  1 04:47 booklover_test.py
-rw------- 1 mxg9xv users    0 Apr  1 04:44 __init__.py

./demo:
total 24
-rw------- 1 mxg9xv users 1416 Apr  1 06:13 booklover.py
-rw------- 1 mxg9xv users    0 Apr  1 04:44 booklover_results.txt
-rw------- 1 mxg9xv users 1782 Apr  

 ls -l -lR
.:
total 20
-rw------- 1 mxg9xv users 1198 Apr  1 04:47 booklover.py
-rw------- 1 mxg9xv users    0 Apr  1 04:44 booklover_results.txt
-rw------- 1 mxg9xv users 1782 Apr  1 04:47 booklover_test.py
-rw------- 1 mxg9xv users    0 Apr  1 04:44 __init__.py

## Task 2

(1 point)

Put the URL of your GitHub repo here. Just paste it into a Markdown cell.

URL: https://github.com/Datascifer/hwm09p

## Task 3

(1 point)

Show the results of installing your package.

```bash
!pip install -e .
```

Defaulting to user installation because normal site-packages is not writeable
Processing /sfs/qumulo/qhome/mxg9xv/hwm09p
  Preparing metadata (setup.py) ... done
Building wheels for collected packages: demo
  Building wheel for demo (setup.py) ... done
  Created wheel for demo: filename=demo-0.1-py3-none-any.whl size=3338 sha256=bd9fb39c32426f47d40d07374ceda70df6d67639f8237680b5a4e834755d2790
  Stored in directory: /tmp/pip-ephem-wheel-cache-v30mahjd/wheels/14/fa/dd/2f71f63ad5c4e0c191dd195959a2c4d14a533ae8e532f1e5fa
Successfully built demo
Installing collected packages: demo
Successfully installed demo-0.1

## Task 4

(1 point)

Create a file outside your repo to test your package by running it.

To do this, import the package into your file and create a BookLover object.

Then add a book and then print number books read.

Then run the file. 

Show the output of running the file below, using a command like the following:

```bash
!python ../book_lover_demo.py
```

In [2]:
# Python 

 python test_package.py test_package.py test_package.py
/home/mxg9xv/.local/lib/python3.11/site-packages/demo/booklover.py:12: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison
  if book_name in self.book_list['book_name'].values:
Book 'The Great Gatsby' added successfully.
Book '1984' added successfully.
Number of books read by Steven Segal: 0

IndentationError: unexpected indent (2189409522.py, line 3)